shiny/Data-Ingest/app.R

```diff
@@ -0,0 +1,76 @@
+#
+# This is a Shiny web application. You can run the application by clicking
+# the 'Run App' button above.
+#
+# Find out more about building applications with Shiny here:
+#
+#    http://shiny.rstudio.com/
+#
+
+library(shiny)
+library(PEcAn.data.land)
+library(shinydashboard)
+library(dataone)
+
+#stopifnot
+
+# Define UI for application
+
+ui <- dashboardPage(
+  
+  dashboardHeader(title = "Data Ingest"),
+  dashboardSidebar(
+    sidebarMenu(
+      menuItem("Import Data", tabName = "importData", icon = icon("file")),
+      menuItem("Step 2 -- dbfiles record", tabName = "step2", icon = icon("cog")),
+      menuItem("Step 3 -- format record", tabName = "step3", icon = icon("cog")),
+      menuItem("Step 4 -- etc.", tabName = "step4", icon = icon("cog"))
+    )
+  ),
+  dashboardBody(
+    tabItems(
+      tabItem(tabName = "importData",
+              fluidRow( 
+                box(
+                  textInput("id", label = h3("Import From DataONE"), placeholder = "Enter doi or id here"),
+                  actionButton(inputId = "D1Button", label = "Upload"),
+                  hr(),
+                  fluidRow(column(12, verbatimTextOutput("identifier"))) 
+                ),
+                
+                box(
+                  # https://github.com/rstudio/shiny-examples/blob/master/009-upload/app.R
+                  fileInput(inputId = "file", label = h3("Upload Local Files"), accept = NULL, multiple = TRUE, placeholder = "Drag and drop files here"),
+                  p("This isn't linked to the server yet")
+                )
+              )
+      ),
+      
+      tabItem(tabName = "step2",
+              h2("dbfiles tab content")
+      )
+      
+      
+    )
+  )
+)
+
+server <- function(input, output) {
+  
+  # d1d <- eventReactive(input$D1Button, { input$id }) #print doi on click
+  
+  # How do I force R to load the dependencies before I run dataone_download? brew, redland, datapack, dataone
+   d1d <- eventReactive(input$D1Button, { PEcAn.data.land::dataone_download(input$id) }) #run dataone_download with input from id on click
```

Comment:
How is dataone_download running without an output directory specified?

Reply:
It defaults to the dbfiles directory path (fs/data1/pecan.data/dbfiles) Should I include the option for the user to change this on the web version? I thought that we were reserving the ability to select the destdir for console users. I can add an output selector option to the dashboard if need be.

Reply:
You should be grabbing the dbfile path from the config.php and then passing it in explicitly. Having it only work on the BU node in the network, and not for any collaborators or users, is a pretty big design limitation. This doesn't require a completely fluid destdir output selector -- that seems dangerous.